src/effects/effectchainslot.cpp

```diff
@@ -245,19 +245,11 @@ EffectChainPointer EffectChainSlot::getEffectChain() const {
 }
 
 void EffectChainSlot::clear() {
-    // Stop listening to signals from any loaded effect
     if (m_pEffectChain) {
-        m_pEffectChain->removeFromEngine(m_pEffectRack->getEngineEffectRack(),
-                                         m_iChainSlotNumber);
-        foreach (EffectSlotPointer pSlot, m_slots) {
+        for (EffectSlotPointer pSlot: m_slots) {
             pSlot->clear();
         }
-        m_pEffectChain->disconnect(this);
-        m_pEffectChain.clear();
```

Comment:
The crash was caused by this line clearing the m_pEffectChain QSharedPointer. When a new effect was loaded, dereferencing this pointer caused a segfault. The other lines needed to be removed as well because the chain needs to be added to the engine for loading the new effect to work.